sunpy/map/mapbase.py

```diff
@@ -726,19 +726,22 @@ def max(self, *args, **kwargs):
     @staticmethod
     def _parse_fits_unit(unit_str):
         replacements = {'gauss': 'G',
-                        'dn': 'ct',
-                        'dn/s': 'ct/s',
                         'counts / pixel': 'ct/pix',}
         if unit_str.lower() in replacements:
             unit_str = replacements[unit_str.lower()]
         unit = u.Unit(unit_str, format='fits', parse_strict='silent')
         if isinstance(unit, u.UnrecognizedUnit):
-            warn_metadata(f'Could not parse unit string "{unit_str}" as a valid FITS unit.\n'
-                          f'See {_META_FIX_URL} for how to fix metadata before loading it '
-                          'with sunpy.map.Map.\n'
-                          'See https://fits.gsfc.nasa.gov/fits_standard.html for '
-                          'the FITS unit standards.')
-            unit = None
+            # NOTE: Special case DN here as it is not part of the FITS standard, but
+            # is widely used and is also a recognized astropy unit
+            if 'DN' in str(unit):
```

Comment:
This still seems like a *very* unreliable/unsatisfying way to check for this, but i cannot come up with any other way. Once you parse the unit using `format='fits'` as we do above, the whole thing just becomes a big blob of `UnrecognizedUnit` so it becomes impossible to decompose each piece into `u.Unit` objects.